shuup/utils/multilanguage_model_form.py

```diff
@@ -125,15 +151,27 @@ def _save_translation(self, instance, translation):
     def save(self, commit=True):
         self._set_fields_for_language(self.default_language)
         self.pre_master_save(self.instance)
+
+        # Save is necessary here since translations can not be
+        # attached to non saved object
         self.instance = self._save_master(commit)
         self._save_translations(self.instance, self.cleaned_data)
+
+        # Save the master once again since the save might involve
+        # some procedures that requires translations like generating
+        # slug from translated name.
+        self._save_master(commit)
         return self.instance
 
     def _set_fields_for_language(self, language):
-        self.instance.set_current_language(language)
-        for field in self.translation_fields:
-            value = self.cleaned_data["%s__%s" % (field.name, language)]
-            field.save_form_data(self.instance, value)
+        original_language = self.instance.get_current_language()
```

Comment:
sorry for the late comment :) Looking at the django-parler docs, we should be able to use a context manager here:
```
from parler.utils.context import switch_language

with switch_language(self.instance, language):
    do stuff
```